src/digests/digest.lisp

```diff
@@ -177,17 +177,18 @@
              (let ((digest-size ,(if single-digest-p
                                      (second (first specs))
                                      `(etypecase state
-                                        ,@(reverse specs)))))
+                                        ,@(reverse specs))))
+                   (state-copy (copy-digest state)))
```

Comment:
It might be a convenient feature, but it would have to be done in a way that minimizes consing.